src/nvim/diff.c

```diff
@@ -1597,6 +1597,34 @@ static bool diff_equal_entry(diff_T *dp, int idx1, int idx2)
   return true;
 }
 
+// Compare the characters at "p1" and "p2".  If they are equal (possibly
+// ignoring case) return true and set "len" to the number of bytes.
+static bool diff_equal_char(const char_u *const p1, const char_u *const p2,
+                            int *const len)
```

Comment:
I know about the objections to `const` pointers but these params should all be `const` pointers to begin with because of the function and param descriptions. It's a new function so nothing should break.